python/lsst/daf/butler/core/_butlerUri/http.py

```diff
@@ -345,18 +263,21 @@ def session(self) -> requests.Session:
                 refreshToken(ButlerHttpURI._session)
             return ButlerHttpURI._session
 
-        baseURL = self.scheme + "://" + self.netloc
-
-        if isWebdavEndpoint(baseURL):
-            log.debug("%s looks like a Webdav endpoint.", baseURL)
-            s = getHttpSession()
-        else:
-            raise RuntimeError(f"Only Webdav endpoints are supported; got base URL '{baseURL}'.")
-
+        s = getHttpSession()
         ButlerHttpURI._session = s
         ButlerHttpURI._sessionInitialized = True
         return s
 
+    @property
+    def is_webdav_endpoint(self) -> bool:
+        """Check if the current endpoint implements WebDAV features""" 
+        try:
+            return ButlerHttpURI._is_webdav
+        except AttributeError:
+            pass
+        ButlerHttpURI._is_webdav = isWebdavEndpoint(self.root_uri())
```

Comment:
Doesn't this mean that every subsequent ButlerURI thinks it's webdav even if it's for a completely different server? I think the cache should be indexed by the root_uri shouldn't it?

Reply:
It might be easier to have a non-class function that uses an lru_cache that takes the `self.root_uri()` as argument. You'll get caching for free (ButlerURI can now be used as a key).

Reply:
I made `_is_webdav` class-wide to be coherent with what I do with `_session`, although I admit I'm unsure this is the best way to do it. In my tests, without using the class scope, `_session` and `_is_webdav` would be refreshed every time we change directory/file, even though it is the same endpoint/server.
I will look into `functools.lru_cache` which I didn't know before.
I'll also fix mypy/lint issues in an upcoming commit.

Reply:
Global cache is absolutely fine but as written it was caching a single value regardless of the URI root. We need a per-root cache which can doesn't need to be attached to the class itself (hence lru_cache decorator on a standalone function).